### ISSUE TITLE
Different initial values for each mcmc chain 

### ISSUE THREAD

**Issue by G-Kazantzidis** [Genentech/jmpost#4]
With the current implementation, the initial values are part of the stanAll object. Since there is only one slot for the initial values, the user is unable to define different initial values for every mcmc run. It would be better if the user is able to define different values.

**Comment by danielinteractive** [comment]
Yeah good point that we should be able to change inits after the model is compiled. Maybe there is a way to do that and still define some default inits based on the model?

**Comment by gowerc** [comment]
This is currently blocked by #91

**Comment by mercifr1** [comment]
Using different initial values for each chain is indeed the 'recommended' way to increase the chance that convergence is achieved at the global minimum.

**Comment by danielinteractive** [comment]
Thanks @mercifr1 , ok then let's take this feature into the initial release

**Comment by danielinteractive** [comment]
It would be good that we have some sensible default initial values which are different for each chain when the user is too lazy to specify any initial values.
Apart from that, if the user does specify initial values, then I would assert that it is the right number of them.

**Comment by danielinteractive** [comment]
yeah I think random draw from prior should usually be ok, maybe shrink it a bit to the mean because if you have a vague prior this could be pretty far off, e.g. 0.5 * mean + 0.5 * random draw

**Comment by danielinteractive** [comment]
I would fail with an error in both cases. Since we want the user to be conscious of the number of chains and to encourage different initial values for the chains

**Comment by danielinteractive** [comment]
Yeah agree, option (3) looks best to me too.

**Comment by gravesti** [comment]
This (3) approach is what came to my mind as well.
